query/src/test/java/com/connexta/replication/query/QueryServiceTest.java

```diff
@@ -0,0 +1,165 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package com.connexta.replication.query;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.replication.api.NodeAdapter;
+import com.connexta.replication.api.data.Filter;
+import com.connexta.replication.api.data.FilterIndex;
+import com.connexta.replication.api.data.Metadata;
+import com.connexta.replication.api.data.QueryRequest;
+import com.connexta.replication.api.data.QueryResponse;
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.queue.SiteQueue;
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryServiceTest {
+
+  private static final long PERIOD = 60L;
+
+  private static final String SITE_ID = "siteid";
+
+  @Mock private Site site;
+
+  @Mock private QueryServiceTools queryServiceTools;
+
+  @Mock private ScheduledExecutorService executor;
+
+  @Mock private SiteQueue siteQueue;
+
+  @Mock private NodeAdapter adapter;
+
+  private QueryService queryService;
+
+  @Before
+  public void setUp() throws Exception {
+    when(queryServiceTools.getQueueFor(site)).thenReturn(siteQueue);
+    when(queryServiceTools.getAdapterFor(site)).thenReturn(adapter);
+    when(queryServiceTools.getPeriod()).thenReturn(PERIOD);
+    queryService = new QueryService(site, queryServiceTools, executor);
+  }
+
+  @Test
+  public void update() {
+    Site site2 = mock(Site.class);
+    queryService.update(site2);
+    verify(queryServiceTools).getAdapterFor(site2);
+  }
+
+  @Test
+  public void start() {
+    ScheduledFuture future = mock(ScheduledFuture.class);
+    when(executor.scheduleAtFixedRate(
+            any(Runnable.class), eq(0L), eq(PERIOD), eq(TimeUnit.SECONDS)))
+        .thenReturn(future);
+    queryService.start();
+    verify(executor)
+        .scheduleAtFixedRate(any(Runnable.class), eq(0L), eq(PERIOD), eq(TimeUnit.SECONDS));
+    assertThat(queryService.future, notNullValue());
+  }
+
+  @Test
+  public void query() throws Exception {
```

Comment:
✏️ You need test cases that will ensure that you do keep the right modified timestamp when dealing with multiple metacards in a single query and a mixture of them.